internal/value/value.go

```diff
@@ -0,0 +1,106 @@
+package value
+
+import (
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/structure"
+)
+
+// Bool evaluates the typed value of the value
+// and coerces to a pointer of a boolean.
+func Bool(rawValue cty.Value) *bool {
+	if rawValue.IsNull() {
+		return nil
+	}
+
+	value := rawValue.True()
+	return &value
+}
+
+// String evaluates the typed value of the value
+// and coerces to a pointer of a string.
+func String(rawValue cty.Value) *string {
+	if rawValue.IsNull() {
+		return nil
+	}
+
+	value := rawValue.AsString()
+	return &value
+}
+
+// Int evaluates the typed value of the value
+// and coerces to a pointer of an int.
+func Int(rawValue cty.Value) *int {
+	if rawValue.IsNull() {
+		return nil
+	}
+
+	int64Value, _ := rawValue.AsBigFloat().Int64()
+	value := int(int64Value)
+	return &value
+}
+
+// Float64 evaluates the typed value of the value
+// and coerces to a pointer of a float64.
+func Float64(rawValue cty.Value) *float64 {
+	if rawValue.IsNull() {
+		return nil
+	}
+
+	value, _ := rawValue.AsBigFloat().Float64()
+	return &value
+}
+
+// Strings evaluates the typed value of the value
+// and coerces to a pointer of a slice of strings.
+func Strings(rawValues cty.Value) *[]string {
+	if rawValues.IsNull() {
+		return nil
+	}
+
+	var value []string
+	for _, rawValue := range rawValues.AsValueSlice() {
+		value = append(value, rawValue.AsString())
+	}
+
+	return &value
+}
+
+// Map evaluates the typed value of the value
+// and coerces to a map[string]interface{}.
+func Map(rawValue cty.Value) map[string]interface{} {
+	if rawValue.IsNull() {
+		return nil
+	}
+
+	m := make(map[string]interface{})
+	for key, value := range rawValue.AsValueMap() {
+		m[key] = value.AsString()
```

Comment:
Is the value always a string here?

Reply:
Indeed, this is always a string 👍🏻 thanks for double checking.

We're going to take a multi step approach in adding these kind of helpers, focusing on the provider first but afterwards try to fix a couple of types in the go SDK as well so we stop overusing `map[string]interface{}` instead of a more accurate `map[string]string`.

I'll mark this PR as Draft for now until we finalize porting all the resources to use them, just to make sure we aren't missing anything and they work as expected.